# Marchantia polymorpha pop gen project

This repository contains the analysis workflow for the manuscript entitled "Population genomics of Marchantia polymorpha subsp. ruderalis reveals evidence of climate adaptation".

The manuscript is available from [Current Biology](https://www.sciencedirect.com/science/article/pii/S0960982225000089).

Abstract
Sexual reproduction results in the development of haploid and diploid cell states during the life cycle. In bryophytes, the dominant multicellular haploid phase produces motile sperm that swim through water to the egg to effect fertilization from which a relatively small diploid phase develops. In angiosperms, the reduced multicellular haploid phase produces non-motile sperm that is delivered to the egg through a pollen tube to effect fertilization from which the dominant diploid phase develops. These different life cycle characteristics are likely to impact the distribution of genetic variation among populations. However, little is known about the distribution of genetic variation among wild populations of bryophytes. To investigate how genetic variation is distributed among populations of a bryophyte and to establish the foundation for population genetics research in bryophytes, we described the genetic diversity of collections of Marchantia polymorpha subsp. ruderalis, a cosmopolitan ruderal liverwort. We identified 78 genetically unique (non-clonal) from a total of 209 sequenced accessions collected from 37 sites in Europe and Japan. There was no detectable population structure among European populations but significant genetic differentiation between Japanese and European populations. By associating genetic variation across the genome with global climate data, we showed that temperature and precipitation influence the frequency of potentially adaptive alleles. This collection establishes the core of an experimental platform that exploits natural genetic variation to answer diverse questions in biology.

The analysis workflow is described [here](03_code/README.md), from which individual workbooks containing code for specific analyses can be accessed.

Workbook:[Genome analyses](03_code/MP.00_genome_analyses.md)  
Workbook:[Mapping](03_code/MP.01_mapping.md)  
Workbook:[Kraken](03_code/MP.02_kraken_contmaination.md)  
Workbook:[DNA damage](03_code/MP.03_dna_damage.md)  
Workbook:[Genome coverage](03_code/MP.04_genome_coverage.md)  
Workbook:[Variant calling and filtering](03_code/MP.05_variant_calling_and_filtering.md)  
Workbook:[Sampling sites](03_code/MP.06_sampling_site_maps_and_data.md)  
Workbook:[PCAs](03_code/MP.07_PCAs.md)  
Workbook:[ANGSD](03_code/MP.08_ANGSD.md)  
Workbook:[NGSadmix](03_code/MP.09_NGSadmix.md)  
Workbook:[Treemix](03_code/MP.10_treemix.md)  
Workbook:[Admixtools](03_code/MP.11_admixtools.md)  
Workbook:[Population history using SMC++](03_code/MP.12_smc++.md)  
Workbook:[Genome-wide genetic variation](03_code/MP.13_genomewide_genetic_variation.md)  


Any reuse of data or code is encouraged with due acknowledgement, either via citation of the published manuscript (when available) and/or GitHub repository. Comments, suggestions, and discussion are welcome.

******
## License
<a rel="license" href="http://creativecommons.org/licenses/by/4.0/"><img alt="Creative Commons Licence" style="border-width:0" src="https://i.creativecommons.org/l/by/4.0/88x31.png" /></a><br />This work is licensed under a <a rel="license" href="http://creativecommons.org/licenses/by/4.0/">Creative Commons Attribution 4.0 International License</a>.
